silhouette/Graphtec.py

```diff
@@ -573,216 +699,196 @@ def setup(s, media=132, speed=None, pressure=None, toolholder=None, pen=None, cu
         if pressure <  1: pressure = 1
         if pressure > 33: pressure = 33
-        if s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
-          s.write(b"FX%d,%d\x03" % (pressure, toolholder));
+        if self.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
+          self.send_command(tool.pressure(pressure))
         else:
-          s.write(b"FX%d\x03" % pressure);
+          self.send_command("FX%d" % pressure)
           # s.write(b"FX%d,0\x03" % pressure);       # oops, graphtecprint does it like this
-        print("pressure: %d" % pressure, file=s.log)
+        print("pressure: %d" % pressure, file=self.log)
 
-      if s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
+      if self.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
         if pen:
-          s.write(b"FC0,1,%d\x03"  % (toolholder))
+          self.send_command(tool.cutter_offset(0, 0.05))
 
-      if s.leftaligned:
-        print("Loaded media is expected left-aligned.", file=s.log)
+      if self.leftaligned:
+        print("Loaded media is expected left-aligned.", file=self.log)
       else:
-        print("Loaded media is expected right-aligned.", file=s.log)
+        print("Loaded media is expected right-aligned.", file=self.log)
 
       # Lift plotter head at sharp corners
-      if s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
-        if sharpencorners:
-          s.write(b"FE1,%d\x03" % toolholder)
-        else:
-          s.write(b"FE0,%d\x03" % toolholder)
+      if self.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
+        self.send_command(tool.lift(sharpencorners))
 
         if pen:
-          s.write(b"FF0,0,%d\x03" % (toolholder))
+          self.send_command(tool.sharpen_corners(0, 0))
         else:
           sharpencorners_start = int((sharpencorners_start + 0.05) * 10.0)
           sharpencorners_end = int((sharpencorners_end + 0.05) * 10.0)
-          s.write(b"FF%d,0,%d\x03FF%d,%d,%d\x03" % (sharpencorners_start, toolholder, sharpencorners_start, sharpencorners_end, toolholder))
+          self.send_command(tool.sharpen_corners(sharpencorners_start, sharpencorners_end))
 
       # robocut/Plotter.cpp:393 says:
       # It is 0 for the pen, 18 for cutting. Default diameter of a blade is 0.9mm
       # C possible stands for curvature. Not that any of the other letters make sense...
       # C possible stands for circle.
       # This value is the circle diameter which is exectuted on direction changes on corners to adjust the blade.
       # Seems to be limited to 46 or 47. Values above does keep the last setting on the device.
-      if s.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
+      if self.product_id() == PRODUCT_ID_SILHOUETTE_CAMEO3:
         if not pen:
-          circle = 0.5 + bladediameter * 20
-          s.write(b"FC0,1,%d\x03FC%d,1,%d\x03" % (toolholder, circle, toolholder))
+          self.send_command([
+            tool.cutter_offset(0, 0.05),
+            tool.cutter_offset(bladediameter + 0.05, 0.05)])
```

Comment:
I think the new dx is differently rounded then the old one:
For instance for bladediameter=0.85 we get:
Old: (0.5+0.85*20)SU=17.5 SU -> as int 17SU
New: (0.85+0.05)*20 SU = 18 SU -> as int still 18SU
If we want to maintain the rounding to the closest number in SUs the correct formula would be `bladediameter + 0.025` correct?
Please correct me, if I don't see why the new way of rounding is superior to the old one.

Reply:
It took my a time to identify the idea behind the 0.5, but now its clear. But it should be `0.0025`

Reply:
Hm, no I think is should be 0.025, as then
```[new code] = (int)((bladediameter + 0.025) * 20) = (int)(0.025 * 20 + bladediameter * 20) = (int)(0.5 + bladediameter * 20) = [old code]```

Reply:
check